### PR TITLE
19_algorithmic_thinking lesson

### PR DESCRIPTION
**Pull request by mgoel283**

### CODE REVIEW

File: lessons/19_algorithmic_thinking/19_algorithmic_thinking.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        for j in range(1, len(nums)):
+            if nums[i] + nums[j] == target:
+                return [i, j]
+```
+Since we have a nested for loop and check all the possible pairs of the list, the runtime complexity of this algorithm is O(n^2).
+
+
+Optimized solution:
+```python
+def two_sum_optimized():
```
Comment (ashiq-r31) [inline lessons/19_algorithmic_thinking/19_algorithmic_thinking.md:123]: nums, target missing in params

File: lessons/19_algorithmic_thinking/19_algorithmic_thinking.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    complements = {}
+    
+    for i in range(len(nums)):
+        complement = target - nums[i]
+        if complement in complements:
+            return (complements[complement], i)
```
Comment (ashiq-r31) [inline lessons/19_algorithmic_thinking/19_algorithmic_thinking.md:129]: this returns a tuple whereas the brute force solution returns a list

**Review by ashiq-r31 — CHANGES_REQUESTED** [review]
Looks good, there a couple things to clean up that i've added comments for.

**Review by ashiq-r31 — APPROVED** [review]

**Review by pab2163 — APPROVED** [review]
looks great!

**Pull request merged**